python/delta_sharing/rest_client.py

```diff
@@ -253,6 +266,15 @@ def _get_internal(self, target: str, data: Optional[Dict[str, Any]] = None):
     def _post_internal(self, target: str, data: Optional[Dict[str, Any]] = None):
         return self._request_internal(request=self._session.post, target=target, json=data)
 
+    def _head_internal(self, target: str):
+        assert target.startswith("/"), "Targets should start with '/'"
```

Comment:
Can we do what other two methods are doing, e.g `return self._request_internal(request=self._session.head, target=target)`

Reply:
`_request_internal` returns only the request content, but we only need the headers from the request, hence a new function